### PR TITLE
OperatorSource reconciliation refactor

### PR DESCRIPTION
**Pull request by aravindhp**
- Extract Status fields into common Phase and ObjectPhase types
- Modularize phase and transitioner logic
- Reuse Phase instead of PhaseNext

No new functionality has been added or modified by this PR.

### CODE REVIEW

File: pkg/operatorsource/factory.go [hunk 1]
```diff
@@ -10,7 +10,9 @@ import (
 
 	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
 	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
-	"github.com/operator-framework/operator-marketplace/pkg/operatorsource/phase"
+	r "github.com/operator-framework/operator-marketplace/pkg/operatorsource/phase"
```
Comment (SamiSousa) [inline pkg/operatorsource/factory.go]: Any reasoning for using the alias `r`? It seems a little confusing. Also, what's the difference between this `phase` package and the new one?

File: pkg/apis/marketplace/v1alpha1/phase_types.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// ObjectPhase describes the phase of a Marketplace object is in along with the
+// last time a phase transition occurred and when the object was last updated
+type ObjectPhase struct {
+	// Current phase of the object
+	Phase `json:"phase,omitempty"`
+
+	// Last time the object has transitioned from one phase to another
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Last time we update the status of the object
```
[the remark this answers is not in the document]
Comment (aravindhp) [reply pkg/apis/marketplace/v1alpha1/phase_types.go]: Fixed

**Pull request merged**